dftimewolf/lib/collectors/virustotal.py

```diff
@@ -0,0 +1,231 @@
+      else:
+        self.ModuleError(f"File not found {download_link}", critical=False)
+
+      if self.vt_type == 'pcap':
+        self.logger.info('Writing pcap to file')
+
+        container = containers.File(
+            name=vt_hash, path=os.path.abspath(filename))
+        self.state.StoreContainer(container)
+        self.logger.info('Finished writing evtx to file')
+
+      if self.vt_type == 'evtx':
+        self.logger.info('Writing evtx to file')
+
+        # Unzip the file so that plaso can go over evtx part in the archive
+        client_output_file = os.path.join(self.output_path, vt_hash)
+        if not os.path.isdir(client_output_file):
+          os.makedirs(client_output_file)
+
+        with zipfile.ZipFile(filename) as archive:
+          archive.extractall(path=client_output_file)
+          self.logger.debug(
+              f'Downloaded file extracted to {client_output_file}')
+
+        container = containers.File(
+            name=vt_hash, path=os.path.abspath(client_output_file))
+        self.state.StoreContainer(container)
+        self.logger.info('Finished writing evtx to file')
+
+  # pylint: disable=arguments-differ,too-many-arguments
+  def SetUp(
+      self,
+      hashes: str,
+      vt_api_key: str,
+      vt_type: str,
+      output_path: str = tempfile.mkdtemp(),
+  ) -> None:
+    """Sets up an Virustotal (VT) collector.
+
+    Args:
+      hashes: Coma seperated strings of hashes
+      vt_api_key: Virustotal Enterprise API Key
+      vt_type: Which file to fetch
+      output_path [optional]: Where to store the downloaded files to
+    """
+
+    self.output_path = self._CheckOutputPath(output_path)
+
+    if not hashes:
+      self.ModuleError("You need to specify at least one hash", critical=True)
+      return
+
+    if not vt_type:
+      self.ModuleError(
+          "You need to specify an vt_type from: pcap, evtx", critical=True)
+      return
+
+    self.vt_type = vt_type
+
+    self.hashes_list = [item.strip() for item in hashes.strip().split(",")]
+
+    if not vt_api_key:
+      self.ModuleError(
+          "You need to specify a Virustotal Enterprise API key",
+          critical=True,
+      )
+      return
+
+    self.client = vt.Client(vt_api_key)
+
+    if self.client is None:
+        self.ModuleError(
+            f'Error creating Virustotal Client instance',
+            critical=True,
+        )
+        return
+
+  def _CheckOutputPath(self, output_path: str = tempfile.mkdtemp()) -> str:
+    """Checks that the output path can be manipulated by the module.
+
+    Args:
+    output_path: Full path to the output directory where files will be
+        dumped.
+
+    Returns:
+        The full path to the directory where files will be dumped.
+    """
+    # Check that the output path can be manipulated
+    if not output_path:
+      return tempfile.mkdtemp()
+    elif os.path.exists(output_path):
+      return output_path
+    elif not os.path.exists(output_path):
```

Comment:
This check is superfluous - if you made it here it's because the conditions above have all evaluated to false, so it's safe to assume `output_path` is defined and that it exists in the filesystem.